### PR TITLE
Add explanations for q_ascend and q_descend

### PR DESCRIPTION
**Pull request by backink**
Add supllementary explainations for q_ascend and q_descend, indicating required memory operations that must be implemented in these functions.

### CODE REVIEW

File: queue.h [hunk 1]
```diff
@@ -207,6 +207,8 @@ void q_sort(struct list_head *head, bool descend);
  *
  * No effect if queue is NULL or empty. If there has only one element, do
  * nothing.
+ * The memory allocated to nodes that are removed must be freed after their
```
Comment (jserv) [inline queue.h:210]: Rephrase as following:
> Memory allocated to removed nodes must be freed.

The latter is slightly more direct.

**Review by jserv — CHANGES_REQUESTED** [review]
Execute `sha1sum queue.h` and update the content of file `scripts/checksums`.

**Comment by backink** [comment]
Ok Thanks for the comment.
I'll look into how to implement this.

**Comment by jserv** [comment]
Close via commit 7a02fb25796fcfffbea7e8406af0ff14e9b4ce02

**Comment by jserv** [comment]
Thank @backink for contributing!

**Pull request closed without merging**